Marlin/src/module/stepper.cpp

```diff
@@ -529,10 +529,10 @@ xyze_int8_t Stepper::count_direction{0};
 // Round up when converting from ns to timer ticks
 #define NS_TO_PULSE_TIMER_TICKS(NS) (((NS) + (NS_PER_PULSE_TIMER_TICK) / 2) / (NS_PER_PULSE_TIMER_TICK))
 
-#define TIMER_SETUP_NS (CYCLES_TO_NS(TIMER_READ_ADD_AND_STORE_CYCLES))
+#define TIMER_SETUP_NS CYCLES_TO_NS(timer_read_add_and_store_cycles)
 
-#define PULSE_HIGH_TICK_COUNT hal_timer_t(NS_TO_PULSE_TIMER_TICKS(_MIN_PULSE_HIGH_NS - _MIN(_MIN_PULSE_HIGH_NS, TIMER_SETUP_NS)))
-#define PULSE_LOW_TICK_COUNT hal_timer_t(NS_TO_PULSE_TIMER_TICKS(_MIN_PULSE_LOW_NS - _MIN(_MIN_PULSE_LOW_NS, TIMER_SETUP_NS)))
+#define PULSE_HIGH_TICK_COUNT hal_timer_t(NS_TO_PULSE_TIMER_TICKS(_min_pulse_high_ns - _MIN(_min_pulse_high_ns, TIMER_SETUP_NS)))
+#define PULSE_LOW_TICK_COUNT hal_timer_t(NS_TO_PULSE_TIMER_TICKS(_min_pulse_low_ns - _MIN(_min_pulse_low_ns, TIMER_SETUP_NS)))
```

Comment:
If we're doing constexpr then `constexpr hal_timer_t PULSE_HIGH_TICK_COUNT = [..];` and similar gets rid of the cast.